src/cdomains/relationDomain.ml

```diff
@@ -0,0 +1,188 @@
+open Prelude
+open Pretty
+open Cil
+
+module type RelVar =
+sig
+  type t
+  val compare : t -> t -> int
+  val of_string : string -> t
+  val to_string : t -> string
+  val  hash : t -> int
+  val equal : t -> t -> bool
+end
+
+module type D2  =
+sig
+  type var
+  type t
+  type marshal
+  val is_bot_env : t -> bool
+  val equal : t -> t -> bool
+  val hash : t -> int
+  val compare : t -> t -> int
+  val show : t -> string
+  val pretty : unit -> t -> doc
+  val printXml : 'a BatInnerIO.output -> t -> unit
+  val name : unit -> string
+  val to_yojson : t -> Printable.json
+  val invariant : Invariant.context -> t -> Invariant.t
+  val arbitrary : unit -> t QCheck.arbitrary
+  val leq : t -> t -> bool
+  val join : t -> t -> t
+  val meet : t -> t -> t
+  val widen : t -> t -> t
+  val narrow : t -> t -> t
+  val pretty_diff : unit -> t * t -> doc
+  val bot : unit -> t
+  val is_bot : t -> bool
+  val top : unit -> t
+  val is_top : t -> bool
+  val vars : t -> var list
+  val add_vars : t -> var list -> t
+  val remove_vars : t -> var list -> t
+  val remove_filter : t -> (var -> bool) -> t
+  val keep_vars : t -> var list -> t
+  val keep_filter : t -> (var -> bool) -> t
+  val forget_vars : t -> var list -> t
+  val assign_exp : t -> var -> exp -> bool -> t
+  val assign_var : t -> var -> var -> t
+  val assign_var_parallel : t -> (var * var) list -> t
+  val assign_var_parallel' : t -> var list -> var list -> t
+  val substitute_exp : t -> var -> exp -> bool -> t
+  val unify: t -> t -> t
+  val tag: t -> int
+  val relift: t -> t
+
+  val marshal: t -> marshal
+
+  val unmarshal: marshal -> t
+end
```

Comment:
Instead of duplicating other interfaces this should include the standard ones (`Printable`, `Lattice`).

Reply:
`D2` includes `Lattice.S` now